### DISCUSSION TITLE
[Vector Database]: compatibility with Elasticsearch

### DISCUSSION THREAD

**Discussion by flefevre** [open-webui/open-webui#7522] [Ideas]
Dear OpenWebui team,
I have seen OpenWebui is compatible with both Chromadb and Milvus.
Since Elasticsearch is very used in AI vector search, I would like to know if ElasticSearch is a vector database that you will be compatible with?
If yes, when? what are the different difficulties to manage?
I would appreciate your comments.

And good thanksgiving to all of us! 

reference: https://www.elastic.co/elasticsearch/vector-database

**Comment by nboehlke** [comment]
Hi, I’d like to join the discussion, as I’m also working on a concept where I’d like to use Elasticsearch. Specifically, the idea is to make individual PDFs searchable, similar to a ChatWithPDF feature.

The interim solution, if Elasticsearch is not available or cannot be used, would be to extract the text from the PDFs beforehand and store it in a PostgreSQL database. However, Elasticsearch remains my preferred solution due to its integrated functionality.

@flefevre 
Maybe I’m wrong, but I believe this thread falls more under Q&A or General. Consider changing the category.
